app/models/manageiq/providers/hawkular/common/event_catcher/runner_mixin.rb

```diff
@@ -0,0 +1,54 @@
+module ManageIQ::Providers::Hawkular::Common::EventCatcher::RunnerMixin
+  extend ActiveSupport::Concern
+
+  def log_handle
+    self.class.log_handle
+  end
+
+  def reset_event_monitor_handle
+    @event_monitor_handle = nil
+  end
+
+  def stop_event_monitor
+    @event_monitor_handle.try(:stop)
+  ensure
+    reset_event_monitor_handle
+  end
+
+  def monitor_events
+    event_monitor_handle.start
+    event_monitor_handle.each_batch do |events|
+      event_monitor_running
+      new_events = events.select { |e| whitelist?(e) }
+      log_handle.debug("#{log_prefix} Discarding events #{events - new_events}") if new_events.length < events.length
+      if new_events.any?
+        log_handle.debug "#{log_prefix} Queueing events #{new_events}"
+        @queue.enq new_events
+      end
+      # invoke the configured sleep before the next event fetch
+      sleep_poll_normal
+    end
+  ensure
+    reset_event_monitor_handle
+  end
+
+  def event_monitor_handle
+    @event_monitor_handle ||= self.class.event_monitor_class.new(@ems)
+  end
+
+  def process_event(event)
+    log_handle.debug "Processing Event #{event}"
+    event_hash = event_to_hash(event, @cfg[:ems_id])
+
+    if blacklist?(event_hash[:event_type])
+      log_handle.debug "#{log_prefix} Filtering blacklisted event [#{event}]"
+    else
+      log_handle.debug "#{log_prefix} Adding ems event [#{event_hash}]"
+      EmsEvent.add_queue('add', @cfg[:ems_id], event_hash)
+    end
+  end
+
+  def blacklist?(event_type)
```

Comment:
`blacklist?` takes an `event_type` and `whitelist?` takes an `event`.  Principle of least surprised suggests that these would take the same parameter type.

Also, a small nitpick.  Past tense makes the names work better as predicate methods: `blacklisted?` and `whitelisted?`  I don't really care if you ignore me on this one, just thought I'd point it out.